pkg/cli/admin/release/git.go

```diff
@@ -68,7 +68,7 @@ func (g *git) ChangeContext(path string) (*git, error) {
 }
 
 func (g *git) Clone(repository string, out, errOut io.Writer) error {
-	cmd := exec.Command("git", "clone", repository, g.path)
+	cmd := exec.Command("git", "clone", "--filter=blob:none", "--bare", repository, g.path)
```

Comment:
This change looks reasonable and it will indeed reduce the disk usage size substantially but we need to ensure that this won't break anything. For example, when git command fails, command executes `git fetch` https://github.com/openshift/oc/blob/0d093afbb0d5b706be03a16d79e8b9b436453df7/pkg/cli/admin/release/git.go#L183, what will happen in that case?

Reply:
Additionally, do we test this against squash commits https://github.com/openshift/oc/blob/0d093afbb0d5b706be03a16d79e8b9b436453df7/pkg/cli/admin/release/git.go#L204

Reply:
In regards to squash commits, I have tested changelogs for 4.14.0->4.15.0, which includes repos that use squash commits. The changelogs produced are identical.